base/acme/src/main/java/org/dogtagpki/acme/database/LDAPDatabase.java

```diff
@@ -432,13 +430,14 @@ private static ACMEOrder loadOrder(LDAPEntry entry)
         }
 
         List<ACMEIdentifier> identifiers = new ArrayList<>();
-        for (String type : IDENTIFIER_TYPES) {
-            attr = entry.getAttribute(ATTR_IDENTIFIER + ";" + type);
-            for (String value : Collections.list(attr.getStringValues())) {
-                identifiers.add(new ACMEIdentifier(type, value));
-            }
+        attr = entry.getAttribute(ATTR_IDENTIFIER);
+        for (String identifier : attr.getStringValueArray()) {
+            String[] parts = StringUtils.split(identifier, ":", 2);
+            String type = parts[0];
+            String value = parts[1];
+            identifiers.add(new ACMEIdentifier(type, value));
         }
-        order.setIdentifiers(identifiers.toArray(new ACMEIdentifier[0]));
+        order.setIdentifiers(identifiers.toArray(new ACMEIdentifier[identifiers.size()]));
```

Comment:
Why this change?  The argument is only needed to fix the return type parameter, and it was written in the way suggested in the official doc: https://docs.oracle.com/javase/7/docs/api/java/util/List.html#toArray(T[]) .

Reply:
OK, in the past the one with 0-size array was less efficient, but apparently that's not the case anymore:
https://stackoverflow.com/questions/174093/toarraynew-myclass0-or-toarraynew-myclassmylist-size
I'll undo this change.

Reply:
Removed.